### PR TITLE
Add system-tests for consul state-store integration

### PR DESCRIPTION
**Pull request by mapuri**
Changes are mostly around modifying the test-infra to take netplugin configuration with consul.

In addition there is a:
- fix in netd.go to correctly setup host-label when netplugin configuration is passed from command-line
- fix in consul driver to trim a leading / in the keys as consul driver rejects the keys otherwise.

### CODE REVIEW

File: systemtests/singlehost/regression_test.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func TestOneHostMultipleNets_regress(t *testing.T) {
 	defer func() {
 		utils.DockerCleanup(t, node1, "myContainer4")
 	}()
-	ipAddress = utils.GetIPAddress(t, node1, "purple-myContainer4")
+	ipAddress = utils.GetIPAddress(t, node1, "purple-myContainer4", "ectd")
```
Comment (erikh) [inline systemtests/singlehost/regression_test.go]: you misspelled "etcd" here.

Comment (mapuri) [inline systemtests/singlehost/regression_test.go]: ah, good catch.

I should be using constants as this is error prone, let me take care of this.

**Comment by mapuri** [comment]
@erikh updated the hard-coded strings with constants. PTAL

**Comment by erikh** [comment]
LGTM

**Pull request merged**